### DISCUSSION TITLE
Cross-platform?

### DISCUSSION THREAD

**Discussion by omgitsalexl** [migueldeicaza/SwiftGodot#103] [Q&A]
Looking over the README, I see the **_Bundling Your Extension_** section detailing building our extension and filling out the library portion for each platform. Please excuse my ignorance or oversight, but I don't see anywhere explaining _how_ to build on/for other platforms such as Windows? 

I know the Swift team made an announcement in 2020 for Swift on Windows support, and the [forums](https://forums.swift.org/c/development/windows/) have some activity to them. But is building with SwiftGodot for Windows/UWP possible at this point? 

Thanks for your time!

**Comment by migueldeicaza** [comment]
SwiftGodot doesn’t have any Apple specific dependencies.  But I don’t generally work on windows or Linux, so it really comes down to people trying it out.

I don’t see anything that would prevent it from working

**Reply by omgitsalexl** [reply]
I appreciate you getting back to me! I will try to carve out some time in the near future to put together some Github build scripts and see how compilation on the Windows platform goes. I will report back as soon as I have something. Thanks again.

**Comment by omgitsalexl** [comment]
Been playing around a bit with trying to compile for the Windows platforms. The build process fails within _DocModel.swift_ due to the regex patterns such as:

`DocModel.swift:186:26: let rxEnumMethodMember = #/\[(enum|method|member) ([\w\.@_/]+)\]/#`

Along with issues with not being able to infer a the _Output_ parameter, which may be related to the regex issues.

`DocModel.swift:412:19: mod = mod.replacing(rxTypeName, with: { x in`

The output from the build action can be found [here](https://pipelines.actions.githubusercontent.com/serviceHosts/cdd2428b-d8bc-413d-8755-8f20865de044/_apis/pipelines/1/runs/7/signedlogcontent/2?urlExpires=2023-09-06T17%3A49%3A23.9388900Z&urlSigningMethod=HMACV1&urlSignature=4AQarj3v5ED5IYJv5ki3Kv8ww2Xi624j%2BsAc9qWPWBk%3D). 

Doing a bit of reading regarding the issues, and it seems the Swift compiler on Windows has some issues with regex literals not being supported? [Found on the Swift Forums](https://forums.swift.org/t/regex-literals-requires-quotation-marks-only-on-windows/63140). I am a bit over my head here, but just wanted to pass along current findings if you had any ideas.

**Comment by mgrider** [comment]
Cross-platform would be one of my primary motivations for using SwiftGodot also.

**Comment by daniele-niero** [comment]
Some years later I'd like to leave my 2 cents based on my recent experience on Windows.

It's great! :)

One exception is the debugging: It starts but crash immediately after stepping in on the first break point. I will ask for advice soon, but the rest has been a blast.

Unfortunately I didn't try it, personally, on other platform, but given the fact that many other users are using Mac and Linux, makes my assume that, at least for desktop, SwiftGodot is well suited for targeting them all.

Swift 6.1
Windows 11
Coding and compiling with VSCode
